### PR TITLE
common.h: Guard definition of IS_ALIGNED

### PR DESCRIPTION
**Pull request by yonsch**
common.h defines the IS_ALIGNED macro, which conflicts with a Zephyr macro of the same name. Add guards to define it only if it is not already defined by Zephyr.

### CODE REVIEW

**Comment by yonsch** [comment]
ref: https://github.com/zephyrproject-rtos/zephyr/pull/67243

**Review by dbaluta — APPROVED** [review]

**Comment by andyross** [comment]
I just saw this come by when reviewing the original IS_ALIGNED commit.  So folks are aware: this is the **Zephyr copy** of SOF used for CI.  If you want to fix SOF you have to submit to https://github.com/thesofproject/sof/ , not here.  Has that not happened?  Pretty sure it's OK per policy for us to cherry pick changes ahead of upstreams, but we shouldn't be doing review in our local clones.

File: src/include/sof/common.h
Comment (andyross) [inline src/include/sof/common.h]: Much better to test for ~~__ZEPHYR__~~ `__ZEPHYR__` (*edit: sorry, markdown tripped me up*) instead of the macro itself.  Otherwise you create heisenbugs where "which version" of the macro gets expanded depends in essentially arbitrary ways  based on the order of header inclusions in your particular file.  In this case it's benign, as the macro is simple and the implementations are surely compatible.  But in general this kind of thing is bad practice.  The preprocessor is dangerous, treat it with care.

**Review by andyross — CHANGES_REQUESTED** [review]
Drop a -1 just to make sure this gets submitted to the right spot first.  Also one nitpick

Committers: feel free to override if everyone is happy that this is in progress and won't get accidentally dropped upstream.

**Comment by nashif** [comment]
please submit upstream in the SOF project.

**Comment by yonsch** [comment]
@nashif @andyross When pushing this to upstream SOF it causes a build failure in their CI because Zephyr does not support that macro yet, and we can't merge it in Zephyr because that would fail our CI.
Possible workarounds are to revert back to `#ifndef IS_ALIGNED`, change the macro's name altogether, exclude SOF from our CI temporarily, or fix it in our downstream first and then sumbit it to upstream.

Cc @lyakh @kv2019i

**Comment by andyross** [comment]
Indeed, this is a pain in the butt.  The point of having a local copy of SOF in the Zephyr github organization for use by its own CI is to treat exactly this problem, preventing build failures caused by API changes.  Note that SOF does exactly the same thing on their side: when you build SOF, you get the specific version of Zephyr SOF targets.  One side or the other needs to cherry pick an unmerged submission.

To repeat: my -1 isn't to prevent this from merging here in "zephyr's sof" at all, it's to make sure it's actually submitted upstream so we know the patch doesn't get dropped.  You can submit and link to this PR with a note like "this won't build until Zephyr PR #whatever is integrated".  Then we can cherry pick it here and merge, then SOF can merge simultaneously with their next bump of the Zephyr version.  Then eventually Zephyr will update SOF and the patch will/should merge cleanly as it's already upstream.

**Comment by kv2019i** [comment]
@yonsch Hmm how about this to SOF upstream first:

```
#ifdef __ZEPHYR__
#include <zephyr/sys/util.h>
#endif

#ifndef IS_ALIGNED
#define IS_ALIGNED(size, alignment) ((size) % (alignment) == 0)
#endif
```

This would make SOF ready for https://github.com/zephyrproject-rtos/zephyr/pull/67243 .

**Comment by yonsch** [comment]
@andyross , does @kv2019i 's proposal solve the issue? Since util.h is always included before the macro definition for Zephyr builds, the use of Zephyr's implementation is guaranteed.

**Review by andyross — COMMENTED** [review]
That works for me, sure.  Again my -1 was just to make sure this is submitted in some form, and that both sides are aligned on the plan.  A submission to this particular project (zephyrproject-rtos/sof) in isolation is likely to be forgotten and dropped.

**Comment by marc-hb** [comment]
Merged upstream in https://github.com/thesofproject/sof/pull/8708, @yonsch please close this one.

**Comment by yonsch** [comment]
@nashif With https://github.com/thesofproject/sof/pull/8708 merged, can we merge sof's upstream to our fork?

**Comment by nashif** [comment]
> @nashif With [thesofproject#8708](https://github.com/thesofproject/sof/pull/8708) merged, can we merge sof's upstream to our fork?

yep, working on that.

**Pull request closed without merging**
